### PR TITLE
Add docs for analytics on the design system

### PR DESCRIPTION
**Pull request by owenatgov**
What it says on the tin. Related to https://github.com/alphagov/govuk-design-system/issues/3822

### CODE REVIEW

File: source/analytics/index.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    - page_h1: H1 of the page
+    - status_code: this only detects if the user is viewing a 404 page by checking if the page H1 is 'Page not found'
+    - page_section: Current website section eg: Styles, Components, Patterns
+    - query_string: Query string attached to the page's URL
+
+### External link click
+
+- Event name: external_link_click
+- Description: When a user clicks an external link and leaves the website
+- Trigger: A user clicking a link without the design system website domain
```
[the remark this answers is not in the document]
Comment (owenatgov) [reply source/analytics/index.html.md.erb:89]: I worry this one might change the meaning. Does the word 'outside' imply that the user's interacting with a link outside the design system website whereas we want to convey that it's when a user interacts with a link that takes them away from the DS website.

I wonder if that phrase is actually better: "A user selecting a link that takes them away from the Design System website". What do you think @seaemsi?

**Comment by owenatgov** [comment]
@seaemsi That should be everything. Can you spare a second look pls?

File: source/analytics/index.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+Google Tag Manager (GTM) lets us collect and configure user data and send it to our GA properties. Through our [GTM container](https://tagmanager.google.com/#/container/accounts/341497290/containers/8094879/workspaces/1000045), we can define events on the website, including:
+
+- what triggers the event
+- what information to collect
+- how to present that information
+
+These events are recorded in GA, where they can be analysed.
+
+GTM uses 3 main concepts to manage event data:
```
Comment (domoscargin) [inline source/analytics/index.html.md.erb:35]: I feel like I have to do some extra parsing to understand what the 3 concepts are.
```suggestion
GTM uses 3 main concepts to manage event data: tags, triggers, and variables.
```

File: source/analytics/index.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+Tags are, functionally, events. The tags let us define:
+
+- what a given event is
+- what data we send to GA when it happens
+Triggers define when an event has happened. For example, a trigger may fire:
```
Comment (domoscargin) [inline source/analytics/index.html.md.erb:41]:
```suggestion

Triggers define when an event has happened. For example, a trigger may fire:
```

File: source/analytics/index.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+- current URL
+- target URL if the user has selected a link
+
+However, we can define our own custom variables. Custom variables use the context of where a trigger happens to define themselves. For example, if your variable is the text contents of an element with a given CSS class, the page where the associated tag or trigger happens is where the variable will look for that element.
+
+Our GTM workspace uses a regex table variable attached to all tags called 'Table - GAID'.  This variable pipes event data to either our live property or preview property, depending on what the page domain is.
+
+## User consent
+
+Both GA and GTM require user consent, either through the cookie banner or the [cookies page](https://design-system.service.gov.uk/cookies/),  to set cookies on the website. You can find the JavaScript for handling analytics consent in [cookie-functions.mjs](https://github.com/alphagov/govuk-design-system/blob/main/src/javascripts/components/cookie-functions.mjs).
```
Comment (domoscargin) [inline source/analytics/index.html.md.erb:60]:
```suggestion
Both GA and GTM require user consent, either through the cookie banner or the [cookies page](https://design-system.service.gov.uk/cookies/), to set cookies on the website. You can find the JavaScript for handling analytics consent in [cookie-functions.mjs](https://github.com/alphagov/govuk-design-system/blob/main/src/javascripts/components/cookie-functions.mjs). 
```

File: source/analytics/index.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+### Previewing your changes
+
+Before submitting a change, you should preview your changes using GTM's debug view to make sure they are working as expected and that you have not impacted any other tags. You can read more in [Google's documentation on previewing and debugging changes](https://support.google.com/tagmanager/answer/6107056?hl=en). Remember to accept cookies and turn off ad blockers when testing your changes.
+
+### Review and publishing process
+
+Once you're happy with your changes, you should submit them for review. You can read more in [Google's documentation on publishing and reviewing in GTM](https://support.google.com/tagmanager/answer/6107163?hl=en).
+
+A developer must review, approve and publish the change. You cannot request a review from a group of GTM users, so use the Slack channel to request a review from the developers on the team once you've submitted your change.
```
[the remark this answers is not in the document]
Comment (owenatgov) [reply source/analytics/index.html.md.erb]: I've amended the sentence slightly and added links to both the main and dev channels

**Review by domoscargin — APPROVED** [review]
Looks good to me!

**Pull request merged**